### ISSUE TITLE
Double check all joins are based on `main_chain = true`

### ISSUE THREAD

**Issue by polarker** [alephium/explorer-backend#208]
Ref: https://github.com/alephium/explorer-backend/pull/199#discussion_r855931556

**Comment by tdroxler** [comment]
We also need to check we don't have the same issue as in #218

**Comment by simerplaha** [comment]
This can be done in parallel with other 'double check' tasks like #206 & #191. Should they be merged into a single task?

- Checks ensure that all queries are optimised. 
- Implement any missing test-cases to ensure things are transactional and implement tests to avoid any future triple checking.

**Comment by polarker** [comment]
@simerplaha yes, please 🙏  I put it here just in case we forgot it.

**Comment by tdroxler** [comment]
I double checked everything, I think we are good to close this issue.
